ui/selectors/selectors.js

```diff
@@ -894,6 +894,63 @@ export const getAnySnapUpdateAvailable = createSelector(
     return [...snapMap.values()].some((value) => value === true);
   },
 );
+
+export const getMemoizedTargetSubjectMetadata = createDeepEqualSelector(
+  getTargetSubjectMetadata,
+  (interfaces) => interfaces,
+);
+
+/**
+ * Get the Snap interfaces from the redux state.
+ *
+ * @param state - Redux state object.
+ * @returns the Snap interfaces.
+ */
+const getInterfaces = (state) => state.metamask.interfaces;
+
+/**
+ * Input selector providing a way to pass a Snap interface ID as an argument.
+ *
+ * @param _state - Redux state object.
+ * @param interfaceId - ID of a Snap interface.
+ * @returns ID of a Snap Interface that can be used as input selector.
+ */
+const selectInterfaceId = (_state, interfaceId) => interfaceId;
+
+/**
+ * Get a memoized version of the Snap interfaces.
+ */
+export const getMemoizedInterfaces = createDeepEqualSelector(
+  getInterfaces,
+  (interfaces) => interfaces,
+);
+
+/**
+ * Get a Snap Interface with a given ID.
+ */
+export const getInterface = createSelector(
+  [getMemoizedInterfaces, selectInterfaceId],
+  (interfaces, id) => interfaces[id],
+);
+
+/**
+ * Get a memoized version of a Snap interface with a given ID
+ */
+export const getMemoizedInterface = createDeepEqualSelector(
+  getInterface,
+  (snapInterface) => snapInterface,
+);
+
+export const getInterfaceContent = createSelector(
```

Comment:
To stay consistent with our coding standards, this should have docstring.

Reply:
I don't find docstrings that useful for selectors since they rely on global state more than inputs 🤔

Reply:
I still find it useful. Especially after a while, when coming back to the certain parts of the code we merged. It can be easier to understand some functionality and purpose of the code, find and navigate through the code or find related functionality elements involved in the same feature.

I know that everybody have their own approach to this. Sometimes we just add a docstring to fullfil the "consistency requirement", but sometimes we add it to explain something and make it easier to understand for everybody. I think we should do it for both reasons.

For example this is very nice and helpful:
![Screenshot 2024-02-20 at 19 35 07](https://github.com/MetaMask/metamask-extension/assets/13301024/19a4f933-dd51-4460-86ed-d775502f3397)

On the other side this looks a bit confusing or even unrelated to Snaps features:
![Screenshot 2024-02-20 at 19 37 01](https://github.com/MetaMask/metamask-extension/assets/13301024/eaadb878-1c3e-4f4a-bea9-510b16a469ff)

Of course I can do a full-text search or find usages, use the other tools, etc. But, when looking at the code and trying to understand something, good docstring can be very helpful. 

Good docstring can explain the purpose, explain the functionality more, leave some space for additional explanation in the future when code is changed, gives the function more integrity from a developer's perspective, etc.

Chances are that we'll onboard new developers who will work on this and might need better/easier understanding and navigation through the code. Also, chances are that somebody else will add bunch of code in between this and make understanding and finding stuff more difficult (this file is huge). Then, chances are that this file will be refactored in the future and these functions split, etc.

These are probably the major reasons why I like to see docstrings added. Minor thing, but still nice to have and can be helpful.

I can approve with or without docstrings added in this case, this is not a blocker.

Reply:
Fair point! @GuillaumeRx do you want to add some docstrings here before we get this merged?

Reply:
Yes I can add some docstrings here